src/Persistence_matrix/include/gudhi/matrix.h

```diff
@@ -0,0 +1,2068 @@
+   * The type of the row depends on the choosen options, see @ref PersistenceMatrixOptions::has_intrusive_rows.
+   * 
+   * @param rowIndex @ref rowindex "Row index" of the row to return: @ref IDIdx for @ref chainmatrix "chain matrices"
+   * or updated @ref IDIdx for @ref boundarymatrix "boundary matrices" if swaps occured.
+   * @return Const reference to the row.
+   */
+  const Row_type& get_row(id_index rowIndex) const;
+  //TODO: there is no particular reason that this method is not available for identifier indexing,
+  // just has to be added to the interface...
+  /**
+   * @brief Only available for @ref boundarymatrix "RU matrices" without @ref Column_indexation_types::IDENTIFIER
+   * indexing. Returns the row at the given @ref rowindex "row index" in \f$ R \f$ if @p inR is true and in \f$ U \f$ if
+   * @p inR is false. The type of the row depends on the choosen options, see
+   * @ref PersistenceMatrixOptions::has_intrusive_rows.
+   *
+   * @param rowIndex @ref rowindex "Row index" of the row to return: updated @ref IDIdx if swaps occured.
+   * @param inR If true, returns the row in \f$ R \f$, if false, returns the row in \f$ U \f$.
+   * @return Const reference to the row.
+   */
+  const Row_type& get_row(id_index rowIndex, bool inR);
+
+  /**
+   * @brief Only available for @ref basematrix "base matrices" without column compression and if
+   * @ref PersistenceMatrixOptions::has_map_column_container is true. Otherwise, see @ref remove_last. Erases the given
+   * column from the matrix. If @ref PersistenceMatrixOptions::has_row_access is also true, the deleted column cells are
+   * also automatically removed from their respective rows.
+   *
+   * @param columnIndex @ref MatIdx index of the column to remove.
+   */
+  void remove_column(index columnIndex);
+  //TODO: rename method to be less confusing.
+  /**
+   * @brief The effect varies depending on the matrices and the options:
+   * - @ref basematrix "base matrix" and @ref boundarymatrix "boundary matrix":
+   *    - @ref PersistenceMatrixOptions::has_map_column_container and @ref
+   *      PersistenceMatrixOptions::has_column_and_row_swaps are true: cleans up maps used for the lazy row swaps.
+   *    - @ref PersistenceMatrixOptions::has_row_access and @ref PersistenceMatrixOptions::has_removable_rows are true:
+   *      assumes that the row is empty and removes it.
+   *    - Otherwise, does nothing.
+   * - @ref boundarymatrix "boundary matrix" with \f$ U \f$ stored: only \f$ R \f$ is affected by the above. If properly
+   *   used, \f$ U \f$ will never have empty rows.
+   * - @ref chainmatrix "chain matrix": only available if @ref PersistenceMatrixOptions::has_row_access and
+   *   @ref PersistenceMatrixOptions::has_removable_rows are true. Assumes that the row is empty and removes it.
+   *
+   * @warning The removed rows are always assumed to be empty. If it is not the case, the deleted row cells are not
```

Comment:
Would calling it `erase_empty_row` make sense?

Reply:
Yes, more than `erase_row` for sure.